src/utils/contexts/CartContext.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState, useContext } from "react";
+
+const CartContext = React.createContext();
+
+const CartContextProvider = ({ children }) => {
+  const [cartItems, setCartItems] = useState([]);
+
+  const addProduct = (item) => {
+    const existing = cartItems.find((f) => f.product.id === item.product.id);
+    if (existing) {
+      existing.quantity += item.quantity;
+      const newList = [
+        ...cartItems.filter((f) => f.product.id !== item.product.id),
+        existing,
+      ];
+      setCartItems(newList);
+    } else {
+      setCartItems([...cartItems, item]);
+    }
+  };
```

Comment:
if the condition in the `if\else` statement is true `setCartItems` will be called and if the condition is false `setCartItems` will be called too

The code can be changed to be linear which is easier to read and use the ternary to define the new data:

```jsx
const addProduct = (item) => {
  const itemInCart = cartItems.find((f) => f.product.id === item.product.id);
  const updatedItem = itemInCart
    ? {
      ...existing,
      quantity: existing.quantity + item.quantity,
    }
    : item;
  const newCart = [
    ...cartItems.filter((f) => f.product.id !== item.product.id),
    updatedItem,
  ];

  setCartItems(newCart);
};
```